core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelBean.java

```diff
@@ -0,0 +1,207 @@
+	public static final String PROP_CLASS = "class"; //$NON-NLS-1$
+	public static final String PROP_SCOPE = "scope"; //$NON-NLS-1$
+	public static final String PROP_DEPENDS_ON = "depends-on"; //$NON-NLS-1$
+	public static final String PROP_INIT_METHOD = "init-method"; //$NON-NLS-1$
+	public static final String PROP_DESTROY_METHOD = "destroy-method"; //$NON-NLS-1$
+	public static final String PROP_FACTORY_METHOD = "factory-method"; //$NON-NLS-1$
+	public static final String PROP_FACTORY_BEAN = "factory-bean"; //$NON-NLS-1$
+	public static final String ARG_TYPE = "type"; //$NON-NLS-1$
+	public static final String ARG_VALUE = "value"; //$NON-NLS-1$
+	public static final String PROP_NAME = "name"; //$NON-NLS-1$
+	public static final String PROP_VALUE = "value"; //$NON-NLS-1$
+	public static final String TAG_PROPERTY = "property"; //$NON-NLS-1$
+	public static final String TAG_ARGUMENT = "argument"; //$NON-NLS-1$
+	public static final String TAG_CONSTRUCTOR_ARG = "constructor-arg"; //$NON-NLS-1$
+
+	/**
+	 * @param parent
+	 * @param underlyingNode
+	 */
+	public CamelBean(AbstractCamelModelElement parent, Node underlyingNode) {
+		super(parent, underlyingNode);
+		setUnderlyingMetaModelObject(new GlobalBeanEIP());
+	}
+	
+	public CamelBean(String name) {
+		super(null, null);
+		setUnderlyingMetaModelObject(new GlobalBeanEIP());
+		setParameter(PROP_CLASS, name);
+	}
+	
+	public String getClassName() {
+		return (String)getParameter(PROP_CLASS);
+	}
+	public void setClassName(String name) {
+		setParameter(PROP_CLASS, name);
+	}
+	public String getScope() {
+		return (String)getParameter(PROP_SCOPE);
+	}
+	public void setScope(String value) {
+		setParameter(PROP_SCOPE, value);
+	}
+	public String getDependsOn() {
+		return (String)getParameter(PROP_DEPENDS_ON);
+	}
+	public void setDependsOn(String value) {
+		setParameter(PROP_DEPENDS_ON, value);
+	}
+	public String getInitMethod() {
+		return (String)getParameter(PROP_INIT_METHOD);
+	}
+	public void setInitMethod(String value) {
+		setParameter(PROP_INIT_METHOD, value);
+	}
+	public String getDestroyMethod() {
+		return (String)getParameter(PROP_DESTROY_METHOD);
+	}
+	public void setDestroyMethod(String value) {
+		setParameter(PROP_DESTROY_METHOD, value);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement#getKind(java.lang.String)
+	 */
+	@Override
+	public String getKind(String name) {
+		// due to the missing EIP as underlying meta model we have to tell AbstractCamelModelElement what
+		// kind the attribute is ... so if we got other than ATTRIBUTE please adapt this methods logic!
+		return NODE_KIND_ATTRIBUTE;
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement#parseAttributes()
+	 */
+	@Override
+	protected void parseAttributes() {
+		// there is no model info for beans so we need to parse them manually
+		String value = parseAttribute(PROP_ID);
```

Comment:
what do you think to extract a method instead of reusing the same variable and duplicating code?